src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveParamAnnotationQuickFix.java

```diff
@@ -57,44 +65,66 @@ public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context,
+        sb.append("'@").append(getShortName(annotationsToRemove.get(0))).append("'");
+        for (int j = 1; j < annotationsToRemove.size(); ++j) {
+            sb.append(", '@").append(getShortName(annotationsToRemove.get(j))).append("'");
+        }
+        return Messages.getMessage("RemoveTheModifierFromParameter", sb.toString(), parameter.getName().toString());
+    }
 
-    protected void removeAnnotations(Diagnostic diagnostic, JavaCodeActionContext context,
-                                     List<CodeAction> codeActions, int parameterIndex,
-                                     String label, List<String> annotationsToRemove) {
+     @Override
+     public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+         final PsiElement node = context.getCoveredNode();
+         final CodeAction toResolve = context.getUnresolved();
+         final PsiClass parentType = getBinding(node);
+         final PsiMethod method = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+         CodeActionResolveData data = (CodeActionResolveData) toResolve.getData();
+         List<String> annotationsToRemove = new ArrayList<>();
+         int paramIndex = 0;
 
-        final PsiElement node = context.getCoveredNode();
-        final PsiClass parentType = getBinding(node);
-        final PsiMethod method = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+         if (data.getExtendedDataEntry(ANNOTATION_TO_REMOVE) instanceof List) {
+             annotationsToRemove = (List<String>) data.getExtendedDataEntry(ANNOTATION_TO_REMOVE);
+         }
+         if (data.getExtendedDataEntry(INDEX_KEY) instanceof Integer) {
+             paramIndex = (Integer) data.getExtendedDataEntry(INDEX_KEY);
+         }
 
-        final PsiParameter parameter = method.getParameterList().getParameter(parameterIndex);
-        final PsiAnnotation[] psiAnnotations = parameter.getAnnotations();
-        final List<PsiAnnotation> psiAnnotationsToRemove = new ArrayList<>();
-        Arrays.stream(psiAnnotations).forEach(a -> {
-            if (annotationsToRemove.stream().anyMatch(m -> m.equals(a.getQualifiedName()))) {
-                psiAnnotationsToRemove.add(a);
-            }
-        });
+         final PsiParameter parameter = method.getParameterList().getParameter(paramIndex);
+         final PsiAnnotation[] psiAnnotations = parameter.getAnnotations();
+         final List<PsiAnnotation> psiAnnotationsToRemove = new ArrayList<>();
 
-        RemoveAnnotationsProposal proposal = new RemoveAnnotationsProposal(label, context.getSource().getCompilationUnit(),
-                context.getASTRoot(), parentType, 0, psiAnnotationsToRemove);
-        CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
-        if (codeAction != null) {
-            codeActions.add(codeAction);
-        }
-    }
+         List<String> finalAnnotationsToRemove = annotationsToRemove;
```

Comment:
I don't think you need to create a new variable here, it makes the code harder to understand.